src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -204,7 +205,28 @@ private class TaskLauncherActor(
     case RecheckIfBackOffUntilReached => OfferMatcherRegistration.manageOfferMatcherStatus()
   }
 
+  @SuppressWarnings(Array("all")) // async/await
   private[this] def receiveTaskLaunchNotification: Receive = {
+    case InstanceOpSourceDelegate.InstanceOpRejected(op, TaskLauncherActor.OfferOperationRejectedTimeoutReason) =>
+      // Send a phony task update to trigger a rescheduling.
+      if (inFlightInstanceOperations.exists(_.instanceId == op.instanceId)) {
+        instanceMap.get(op.instanceId).foreach { instance =>
+          import org.apache.mesos.Protos
+          val (taskId, _) = instance.tasksMap.head
+          val state = Protos.TaskState.TASK_FAILED
```

Comment:
I don't like this fake `TASK_FAILED` **at all**. This is Mesos task status and we shouldn't generate it ourselves.